### PR TITLE
Adds a halberd

### PR DESCRIPTION
**Pull request by NightmareChamillian**
## About the PR 
PR for the halberd polearm I've been working on. I plan to add this to surplus crates and/or as apart of a medieval themed traitor kit, but Aloe gave me the go-ahead to PR as a standalone, not-fully-implemented weapon. 

https://www.youtube.com/watch?v=oEvJbO2DaJ4

## Why's this needed? 

melee weapons are sorely lacking in the game, and I want town guard RP, dammit.

### CODE REVIEW

**Comment by TobleroneSwordfish** [comment]
Halt!

**Comment by Sord213** [comment]
long pokey stick

**Comment by Tarmunora** [comment]
Not terribly enthused about "heehoo medieval weaponry in space", personally.

**Comment by ZeWaka** [comment]
I personally don't think anything should be merged if it's not attainable*
might as well be commented-out code that'll rot and get bugs 

**exceptions: exclusively admin gimmick items, or part of a large in-progress project (flock)*

File: code/obj/item/misc_weapons.dm [hunk 1]
```diff
@@ -1717,6 +1717,122 @@ obj/item/whetstone
...
+	stamina_damage = 10
+	stamina_cost = 7
+
+
+	hit_type = DAMAGE_CUT
+	flags = FPRINT | TABLEPASS | USEDELAY | ONBACK
+	c_flags = EQUIPPED_WHILE_HELD
+	item_function_flags = USE_INTENT_SWITCH_TRIGGER | USE_SPECIALS_ON_ALL_INTENTS
+
+	var/guard = null
```
Comment (frawhst) [inline code/obj/item/misc_weapons.dm:1747]: Add a code doc comment (`///` on line above or `//!` on same line) explaining what this var is for

File: code/obj/item/misc_weapons.dm [hunk 2]
```diff
@@ -1717,6 +1717,122 @@ obj/item/whetstone
...
+				icon_state = "halberd"
+				hit_type = DAMAGE_CUT
+				src.click_delay = COMBAT_CLICK_DELAY * 1.25
+				hitsound =  'sound/impact_sounds/Blade_Small_Bloody.ogg'
+				src.setItemSpecial(/datum/item_special/swipe)
+				boutput(user, "<span class='notice'>You will now make heavy swings with the axe!</span>")
+
+
+		user.update_inhands()
+		src.buildTooltipContent()
```
Comment (frawhst) [inline code/obj/item/misc_weapons.dm:1815]:
```suggestion
		src.tooltip_rebuild = TRUE
```
This updates the tooltip 'lazily', so we don't actually update the text unless someone tries to check it.

File: code/obj/item/misc_weapons.dm [hunk 3]
```diff
@@ -1717,6 +1717,122 @@ obj/item/whetstone
...
+				hitsound = 'sound/impact_sounds/Flesh_Stab_1.ogg'
+				src.setItemSpecial(/datum/item_special/rangestab)
+				boutput(user, "<span class='notice'>You will thrust with the tip!</span>")
+
+			if("grab") //attack with the spur on the back
+				force = 10
+				stamina_damage = 40
+				stamina_cost = 20
+				item_state = "halberd1"
+				icon_state = "halberd2"
```
Comment (frawhst) [inline code/obj/item/misc_weapons.dm:1794]: Can these be renamed to `halberd_grab` etc? The numbers are confusing especially when they don't line up.

File: code/obj/item/misc_weapons.dm [hunk 4]
```diff
@@ -1717,6 +1717,122 @@ obj/item/whetstone
...
+	proc/stat_reset() //sets it to normal
+		src.force = 24
+		src.stamina_damage = 10
+		src.stamina_cost = 7
+		src.item_state = "halberd1"
+		src.hit_type = DAMAGE_CUT
+		src.click_delay = COMBAT_CLICK_DELAY * 0.75
+		src.hitsound =  'sound/impact_sounds/Blade_Small_Bloody.ogg'
+		src.setItemSpecial(/datum/item_special/simple)
+		src.buildTooltipContent()
```
Comment (frawhst) [inline code/obj/item/misc_weapons.dm:1835]: same as other thing

File: code/obj/item/misc_weapons.dm [hunk 5]
```diff
@@ -1717,6 +1717,122 @@ obj/item/whetstone
...
+	item_state = "halberd1"
+
+	w_class = W_CLASS_BULKY
+	two_handed = 1
+	throw_range = 10
+	throwforce = 30 //yeet like spear
+	stamina_crit_chance = 5
+
+	//these combat variables change depending on intent- starts with help intent vars
+	force = 28
```
Comment (frawhst) [inline code/obj/item/misc_weapons.dm:1737]: A bunch of these should be defines, pretty much anything which is repeated. Don't need to define every thing on every mode but if you're referencing a number or string on the same mode more than once make it into a define.
Just defining the defaults is Probably Fine
(case in point, this is 28 even though the comment indicates it should be 24)

(syntax: just put `#define HALBERD_STAT_THING 420` above the type definition and `#undef HALBERD_STAT_THING_420` below all the procs.)

**Review by frawhst — COMMENTED** [review]
I'm fine with adding this, I enjoy some of the medieval/renaissance weapons for when we do Woodstation and similar.

File: code/obj/item/misc_weapons.dm [hunk 6]
```diff
@@ -1717,6 +1717,148 @@ obj/item/whetstone
...
+	icon_state = "halberdnormal"
+
+	w_class = W_CLASS_BULKY
+	two_handed = 1
+	throw_range = 10
+	throwforce = 30 //yeet like spear
+	stamina_crit_chance = 5
+
+
+
```
Comment (frawhst) [inline code/obj/item/misc_weapons.dm:1748]:
```suggestion

```
Too much whitespace

File: code/obj/item/misc_weapons.dm [hunk 7]
```diff
@@ -1717,6 +1717,148 @@ obj/item/whetstone
...
+	//these combat variables change depending on intent- starts with help intent vars
+	force = HALB_MED_DAMAGE
+	stamina_damage = HALB_LIGHT_STAMDAM
+	stamina_cost = HALB_LIGHT_STAMCOST
+	var/guard = null //! used to keep track of what melee properties we're using
+
+
```
Comment (frawhst) [inline code/obj/item/misc_weapons.dm:1755]:
```suggestion
```

File: code/obj/item/misc_weapons.dm [hunk 8]
```diff
@@ -1717,6 +1717,148 @@ obj/item/whetstone
...
+	hit_type = DAMAGE_CUT
+	flags = FPRINT | TABLEPASS | USEDELAY | ONBACK
+	c_flags = EQUIPPED_WHILE_HELD
+	item_function_flags = USE_INTENT_SWITCH_TRIGGER | USE_SPECIALS_ON_ALL_INTENTS
+
+
+
```
Comment (frawhst) [inline code/obj/item/misc_weapons.dm:1762]:
```suggestion
```

File: code/obj/item/misc_weapons.dm [hunk 9]
```diff
@@ -1717,6 +1717,148 @@ obj/item/whetstone
...
+	New()
+		..()
+		BLOCK_SETUP(BLOCK_SWORD)
+
+	setupProperties()
+		. = ..()
+		setProperty("deflection", 60)
+		setProperty("block", 40)
+
```
Comment (frawhst) [inline code/obj/item/misc_weapons.dm:1771]:
```suggestion
```

File: code/obj/item/misc_weapons.dm [hunk 10]
```diff
@@ -1717,6 +1717,148 @@ obj/item/whetstone
...
+				stamina_cost = HALB_HEAVY_STAMCOST
+				item_state = "halberdhoriz"
+				icon_state = "halberdnormal"
+				hit_type = DAMAGE_CUT
+				src.click_delay = COMBAT_CLICK_DELAY * 1.25
+				hitsound =  'sound/impact_sounds/Blade_Small_Bloody.ogg'
+				src.setItemSpecial(/datum/item_special/swipe)
+				boutput(user, "<span class='notice'>You will now make heavy swings with the axe!</span>")
+
+
```
Comment (frawhst) [inline code/obj/item/misc_weapons.dm:1827]:
```suggestion
```

**Review by frawhst — APPROVED** [review]
just some whitespace stuff, you rarely need more than 1 empty line in a row
otherwise lgtm

**Pull request merged**